mcs/class/corlib/Mono/RuntimeStructs.cs

```diff
@@ -0,0 +1,34 @@
+//
+// Mono runtime native structs surfaced to managed code.
+//
+// Authors:
+//   Aleksey Kliger <aleksey@xamarin.com>
+//   Rodrigo Kumpera <kumpera@xamarin.com>
+//
+// Copyright 2016 Dot net foundation.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Mono {
+	internal class RuntimeStructs {
+		// class-internals.h MonoRemoteClass
+		[StructLayout(LayoutKind.Sequential)]
+		internal unsafe struct RemoteClass {
+			internal IntPtr default_vtable;
+			internal IntPtr xdomain_vtable;
+			internal MonoClass* proxy_class;
+			internal IntPtr proxy_class_name;
+			internal UInt32 interface_count;
+			// FIXME: How to represent variable-length array struct member?
```

Comment:
How about a private IntPtr for the first address of the variable area plus a bunch of properties/functions to fetch all elements?